### ISSUE TITLE
Incomplete test coverage in Math and Geometry

### ISSUE THREAD

**Issue by marcusnaslund** [magic-lang/ooc-kean#492]
Several public functions and properties are never tested for the covers in `source/math`. Because a few bugs were discovered fixing missing tests for other `Math` classes (see #491 and #478) this should really be done.

- [x] `FloatConvexHull2D`
~~transform~~ (not used)

- [x] `FloatVectorList`
convolve, ~~getWaveletTransform~~ (not used), ~~forwardGaussianKernel~~ (not used), ~~backwardGaussianKernel~~ (not used)

- [x] `FloatComplexVectorList`
~~fastFourierTransformInto~~ (not used)

- [x] `Quaternion`
hamiltonProduct

EDIT (2016-03-02): Removed six finished items and updated the list with what's left.

**Comment by marcusnaslund** [comment]
Complete enough coverage of `Collections` thanks to #494

**Comment by marcusnaslund** [comment]
Updated with status after #491

**Comment by sebastianbaginski** [comment]
What's the status of this ?
Also, isn't it too complex for a single issue ?

**Comment by marcusnaslund** [comment]
I've been updating it, so the description is still valid. 
I don't know, I guess maybe a separate issue for `FloatRotation3D`, but the others are generally missing quite few things, so this issue isn't *really* very big.

**Comment by sebastianbaginski** [comment]
OK, I'm just looking through the issues. Is someone working on it already ? Maybe you have a PR pending ?

**Comment by marcusnaslund** [comment]
No, if I do anything I assign myself to that issue. Feel free to get going. :)

**Comment by sebastianbaginski** [comment]
Thanks, but I have a lot of other stuff to do :) Btw. isn't the "no new code" rule still valid ?

**Comment by marcusnaslund** [comment]
I thought you were above the law. :)

**Comment by sebastianbaginski** [comment]
That was only few lines of code, I swear.
Can we try to make a list of what is missing for this issue to be closed ?
- FloatRotation3D
- ...

**Comment by marcusnaslund** [comment]
> Can we try to make a list of what is missing for this issue to be closed ?

Alright, I'm on it.

**Comment by marcusnaslund** [comment]
Heh, this issue suddenly looks a lot bigger. I will divide it somehow.

**Comment by marcusnaslund** [comment]
Updated with status after #1450

**Comment by marcusnaslund** [comment]
Updated again. Closing this now and will be creating a new issue once I do a new test coverage sweep.
